### ISSUE TITLE
Custom Target Error: 'symbol already defined in module'

### ISSUE THREAD

**Issue by obiwanjacobi** [z88dk/z88dk#1035]
**Warning**: Custom Target Alert!

I have a situation where constants defined in an .asm file are included multiple times when application (C) code refers to it (indirectly). Problem is I don't know why - how that works.

The[ .asm file in question](https://github.com/obiwanjacobi/Zalt/blob/master/Source/z88dk_target/libsrc/_DEVELOPMENT/target/zalt/system/io_addresses.asm) has several `defc name = val` statements which it exports using the `public` directive. For good measure I have surrounded the content with a `IF !_filename_
defc _filename_ = 1` - `ENDIF` which did not fix it.

It has been more than a year since I last worked on my project and to be honest, I'm a little rusty on the inner workings of the z88dk... Perhaps that is a factor also.

Marc

**Comment by aralbrec** [comment]
Yes that's the way to go.  One file declares as public and everyone else imports with extern
